### PR TITLE
[MRG] update dependencies in setup.py and environment.yml

### PR DESCRIPTION
**Pull request by ctb**
Update snakemake &c dependencies in conda and Python install.

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review & merge @taylorreiter @bluegenes

File: setup.py [hunk 1]
```diff
@@ -40,8 +40,8 @@
         "pytest-runner",
     ],
     use_scm_version={"write_to": "genome_grist/version.py"},
-    install_requires=["snakemake>=5.10", "click>=7", "lxml==4.6.1", "pandas"],
-    tests_require=["pytest>=5.1.2", "pytest-dependency"],
+    install_requires=["snakemake>=5.10", "click>=7,<8", "lxml==4.6.1",
```
Comment (taylorreiter) [inline setup.py:43]: should `snakemake>=5.10` be changed to `snakemake=6.4.1` to match `environment.yml`?

File: environment.yml [hunk 2]
```diff
@@ -5,5 +5,7 @@ channels:
  - defaults
 dependencies:
  - mamba
- - screed>=1.0.5,<2
- - sourmash>=4.1.1,<5
```
Comment (taylorreiter) [inline environment.yml:9]: crazy that sourmash isn't needed in the base env...did a scan through the snakefile tho and none of the rules need sourmash outside of conda env `env/sourmash.yml`!

**Review by taylorreiter — COMMENTED** [review]
one question about snakemake version in `setup.py`, otherwise LGTM

File: setup.py (hunk 1 above)
Comment (ctb) [reply setup.py:43]: ooh, good catch, thank you :)

Comment (ctb) [reply setup.py:43]: fixed in ac89910

File: environment.yml (hunk 2 above)
Comment (ctb) [reply environment.yml:9]: yep! I could actually remove lxml, too, by providing it in a conda environment; but I decided not to bother for this PR.

**Review by taylorreiter — APPROVED** [review]
LGTM!

**Pull request merged**